### PR TITLE
Add default implementations in UserDetails

### PR DESCRIPTION
**Pull request by ahmd-nabil**
Closes gh-14275

### CODE REVIEW

File: core/src/main/java/org/springframework/security/core/userdetails/UserDetails.java
Comment (jzheaux) [inline core/src/main/java/org/springframework/security/core/userdetails/UserDetails.java:2]: @ahmd-nabil the Acegi copyright headers are a special case. Will you please leave those as they are?

**Review by jzheaux — CHANGES_REQUESTED** [review]
Thanks, @ahmd-nabil, that was quick! I have just one inline comment for you before merging.

File: core/src/main/java/org/springframework/security/core/userdetails/UserDetails.java
Comment (ahmd-nabil) [reply core/src/main/java/org/springframework/security/core/userdetails/UserDetails.java:2]: sure

Comment (ahmd-nabil) [reply core/src/main/java/org/springframework/security/core/userdetails/UserDetails.java]: @jzheaux done, thanks for your great support btw. I am waiting for your review on PR #14265 whenever you have time.

**Pull request merged**
